src/cmd/tools/read_commitlog/main/main.go

```diff
@@ -96,11 +93,81 @@ func main() {
 
 	runTime := time.Since(start)
 
-	if err := reader.Close(); err != nil {
-		log.Fatalf("unable to close reader: %v", err)
-	}
-
 	fmt.Printf("\nRunning time: %s\n", runTime)                          // nolint: forbidigo
 	fmt.Printf("%d entries read\n", entryCount)                          // nolint: forbidigo
 	fmt.Printf("Total annotation size: %d bytes\n", annotationSizeTotal) // nolint: forbidigo
 }
+
+func summaryAction(reader *filteringReader, top *int) {
+	var (
+		entryCount        uint32
+		start             = time.Now()
+		datapointCount    = map[ident.ID]uint32{}
+		totalIDSize       uint64
+		earliestDatapoint xtime.UnixNano
+		oldestDatapoint   xtime.UnixNano
+	)
+
+	for entry, found := reader.Read(); found; entry, found = reader.Read() {
+		series := entry.Series
+
+		if earliestDatapoint == 0 || earliestDatapoint > entry.Datapoint.TimestampNanos {
+			earliestDatapoint = entry.Datapoint.TimestampNanos
+		}
+		if oldestDatapoint == 0 || oldestDatapoint < entry.Datapoint.TimestampNanos {
+			oldestDatapoint = entry.Datapoint.TimestampNanos
+		}
+
+		datapointCount[series.ID]++
+
+		entryCount++
+	}
+
+	runTime := time.Since(start)
+
+	fmt.Printf("\nRunning time: %s\n", runTime)                                              // nolint: forbidigo
+	fmt.Printf("%d entries read\n", entryCount)                                              // nolint: forbidigo
+	fmt.Printf("time range [%s:%s]\n", earliestDatapoint.String(), oldestDatapoint.String()) // nolint: forbidigo
+
+	datapointCountArr := idPairList{}
+	sizeArr := idPairList{}
+	for ID, count := range datapointCount {
+		IDSize := len(ID.Bytes())
+		totalIDSize += uint64(IDSize)
+		datapointCountArr = append(datapointCountArr, idPair{ID: ID, Value: count})
+		sizeArr = append(sizeArr, idPair{ID: ID, Value: uint32(IDSize)})
+	}
+
+	sort.Sort(sort.Reverse(datapointCountArr))
+	sort.Sort(sort.Reverse(sizeArr))
+
+	fmt.Printf("total ID size: %d bytes\n", totalIDSize)                  // nolint: forbidigo
+	fmt.Printf("total distinct number of IDs %d \n", len(datapointCount)) // nolint: forbidigo
+
+	limit := len(datapointCountArr)
+	if *top > 0 {
+		limit = *top
+	}
+	fmt.Printf("ID datapoint counts: \n") // nolint: forbidigo
+	for i := 0; i < limit; i++ {
+		pair := datapointCountArr[i]
+		fmt.Printf("%-10d %s\n", pair.Value, pair.ID.String()) // nolint: forbidigo
+	}
+
+	fmt.Printf("ID sizes(bytes): \n") // nolint: forbidigo
+	for i := 0; i < limit; i++ {
+		pair := sizeArr[i]
+		fmt.Printf("%-10d %s\n", pair.Value, pair.ID.String()) // nolint: forbidigo
+	}
+}
+
+type idPair struct {
+	ID    ident.ID
+	Value uint32
+}
+
+type idPairList []idPair
```

Comment:
Nit - let's stay away from Java terminology:
```suggestion
type idPairs []idPair
```